src/components/Button/Button.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import styles from "./Button.module.css";
+
+interface ButtonProps {
+  text: string;
+  color?: "default" | "disabled";
+  size?: "large" | "small";
+  width?: string;
+  onClick?: () => void;
+  disabled?: boolean;
+}
+
+export const Button: React.FC<ButtonProps> = ({
+  text,
+  color = "default",
+  size = "large",
+  width,
+  onClick,
+  disabled = false,
+}) => {
```

Comment:
필요한 props들을 type으로 잘 선언해주셨어요!

그런데 만약 버튼 element에 들어가는 많은 props가 다 필요하다면?
이부분을 모두 추가하기엔 너무 어려울것같아요.
그래서 React에서 타입을 선언해준 것들이 있답니다.

```suggestion
interface ButtonProps extends React. ComponentPropsWithRef <'button'> {
  text: string;
  color?: "default" | "disabled";
  size?: "large" | "small";
  width?: string;
}

export const Button: React.FC<ButtonProps> = ({
  text,
  color = "default",
  size = "large",
  width,
  onClick,
  disabled = false,
}) => {
```
`ComponentProps`,`ComponentPropsWithRef `,`ComponentPropsWithoutRef` 가 바로 그것인데요. 이것들을 사용하면 기본 태그 element들의 props를 가져다 사용할 수 가 있어요.

이 셋의 차이나 사용법은 조금 더 찾아보시고 활용하시면 더 간결하게 코드를 작성할 수 있습니다!